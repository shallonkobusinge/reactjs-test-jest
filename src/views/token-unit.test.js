
import React from 'react';
import { render } from '@testing-library/react';
import GenerateToken from './generateToken';

describe('Generate token module', () => {
    it('matches snapshot', () => {

        const { container } = render(<GenerateToken />);
        expect(container.cloneNode(true)).toMatchSnapshot();
    });
})