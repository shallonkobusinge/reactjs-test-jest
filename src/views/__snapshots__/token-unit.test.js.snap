// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate token module matches snapshot 1`] = `
<div>
  <div
    class="login-form-container mt-24"
  >
    <div
      class="text-center font-bold app-color uppercase text-xl header-reg "
    >
      Electricity Token Generation
    </div>
    <form
      novalidate=""
    >
      <div
        class="schoolmanager-container w-full max-w-xs mx-auto"
      >
        <div
          class="grid grid-cols-6 grid-rows-2  gap-8"
        >
          <div
            class="col-span-6 gap-6 sm:col-span-6 sm:row-span-1 "
            data-testid="username"
          >
            <div
              class="input-container"
            >
              <label
                class="block"
                for="username"
                id="usernameLabel"
              >
                Enter your username
              </label>
              <input
                aria-label="Enter your username"
                autocomplete="off"
                class="w-full border input-text mt-3 border-gray-400	rounded appearance-none  py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:ring-2 focus:ring-blue-700 focus:border-transparent"
                name="username"
                placeholder="Enter your username"
                required=""
                step=""
                type="text"
                value=""
              />
              <span
                class="error"
                id="usernameError"
              />
            </div>
          </div>
          <div
            class="col-span-6 gap-6 sm:col-span-6 sm:row-span-1 "
            data-testid="amount"
          >
            <div
              class="input-container"
            >
              <label
                class="block"
                for="amount"
                id="amountLabel"
              >
                Enter amount
              </label>
              <input
                aria-label="Enter amount"
                autocomplete="off"
                class="w-full border input-text mt-3 border-gray-400	rounded appearance-none  py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:ring-2 focus:ring-blue-700 focus:border-transparent"
                name="amount"
                placeholder="Money"
                required=""
                step="any"
                type="number"
                value=""
              />
              <span
                class="error"
                id="amountError"
              />
            </div>
          </div>
          <div
            class="col-span-6 gap-6 sm:col-span-6 sm:row-span-1 "
            data-testid="meter"
          >
            <div
              class="input-container"
            >
              <label
                class="block"
                for="meter"
                id="meterLabel"
              >
                meter
              </label>
              <input
                aria-label="meter"
                autocomplete="off"
                class="w-full border input-text mt-3 border-gray-400	rounded appearance-none  py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:ring-2 focus:ring-blue-700 focus:border-transparent"
                name="meter"
                placeholder="meter"
                required=""
                step=""
                type="text"
                value=""
              />
              <span
                class="error"
                id="meterError"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="mt-8 text-center"
      >
        <button
          class="bg-yellow-500 text-gray-800 font-bold  py-2 px-10 rounded inline-flex items-center submit"
          data-testid="submit-btn"
        >
          <span
            class="text-white"
          >
            Generate
          </span>
        </button>
      </div>
    </form>
    <div
      class="Toastify"
    />
  </div>
</div>
`;
